kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -70,45 +81,64 @@ class SimplifyRuleLHS rule where
 instance SimplifierVariable variable => SimplifyRuleLHS (RulePattern variable)
   where
     simplifyRuleLhs rule@(RulePattern _ _ _ _ _ _) = do
-        let lhsPredicate =
-                makeAndPredicate
-                    requires
-                    (makeCeilPredicate_ left)
-            definedLhs =
-                Conditional.withCondition left
-                $ Condition.fromPredicate lhsPredicate
-        simplifiedTerms <- Pattern.simplifyAndRemoveTopExists definedLhs
+        let lhsWithPredicate = Pattern.fromTermLike left
+        simplifiedTerms <- Pattern.simplifyAndRemoveTopExists lhsWithPredicate
         fullySimplified <-
-            simplifyConditionsWithSmt lhsPredicate simplifiedTerms
+            simplifyConditionsWithSmt
+                (Condition.toPredicate Condition.topTODO)
+                simplifiedTerms
         let rules =
                 map (setRuleLeft rule) (MultiOr.extractPatterns fullySimplified)
         return (MultiAnd.make rules)
       where
-        RulePattern {left, requires} = rule
+        RulePattern {left} = rule
 
         setRuleLeft
             :: RulePattern variable
             -> Pattern variable
             -> RulePattern variable
         setRuleLeft
-            rulePattern
+            rulePattern@RulePattern {requires}
             Conditional {term, predicate, substitution}
           =
             RulePattern.applySubstitution
                 substitution
                 rulePattern
                     { RulePattern.left = term
-                    , RulePattern.requires = predicate
+                    , RulePattern.requires =
+                        Predicate.coerceSort (termLikeSort term)
+                        $ makeAndPredicate predicate requires
                     }
 
+instance SimplifyRuleLHS (RewriteRule Variable) where
+    simplifyRuleLhs =
+        fmap (fmap RewriteRule) . simplifyRuleLhs . getRewriteRule
+
 instance SimplifyRuleLHS (OnePathRule Variable) where
-    simplifyRuleLhs = fmap (fmap OnePathRule) . simplifyRuleLhs . getOnePathRule
+    simplifyRuleLhs =
+        fmap (fmap OnePathRule) . simplifyClaimRule . getOnePathRule
 
 instance SimplifyRuleLHS (AllPathRule Variable) where
-    simplifyRuleLhs = fmap (fmap AllPathRule) . simplifyRuleLhs . getAllPathRule
+    simplifyRuleLhs =
+        fmap (fmap AllPathRule) . simplifyClaimRule . getAllPathRule
 
 instance SimplifyRuleLHS (ReachabilityRule Variable) where
     simplifyRuleLhs (OnePath rule) =
         (fmap . fmap) OnePath $ simplifyRuleLhs rule
     simplifyRuleLhs (AllPath rule) =
         (fmap . fmap) AllPath $ simplifyRuleLhs rule
+
+simplifyClaimRule
+    :: (MonadSimplify simplifier, SimplifierVariable variable)
+    => RulePattern variable
+    -> simplifier (MultiAnd (RulePattern variable))
+simplifyClaimRule rule@(RulePattern _ _ _ _ _ _) =
+    simplifyRuleLhs rule
+        { RulePattern.left =
```

Comment:
Why does this go into `left`, instead of adding the `\ceil` predicate to `requires`?